makefile

```diff
@@ -71,13 +71,13 @@ profiled:
 
 #make a single object profiled library
 profiled_single:
-	perl gen.pl
-	$(CC) $(LTM_CFLAGS) -fprofile-arcs -DTESTING -c mpi.c -o mpi.o
-	$(CC) $(LTM_CFLAGS) -DTESTING -DTIMER demo/timing.c mpi.o -lgcov -o timing
+	cat *mp_*.c > mp_all.c
```

Comment:
how about

```
diff --git a/makefile b/makefile
index 7889c5a..dab28c6 100644
--- a/makefile
+++ b/makefile
@@ -72,5 +72,4 @@ profiled:
 #make a single object profiled library
-profiled_single:
-       cat *mp_*.c > mp_all.c
-       $(CC) $(LTM_CFLAGS) -fprofile-arcs -DTESTING -c mp_all.c -o mp_all.o
+profiled_single: pre_gen
+       $(CC) $(LTM_CFLAGS) -fprofile-arcs -DTESTING -c pre_gen/mp_all.c -o mp_all.o
        $(CC) $(LTM_CFLAGS) -DTESTING -DTIMER demo/timing.c mp_all.o -lgcov -o timing
@@ -78,3 +77,3 @@ profiled_single:
        rm -f *.o timing
-       $(CC) $(LTM_CFLAGS) -fbranch-probabilities -DTESTING -c mp_all.c -o mp_all.o
+       $(CC) $(LTM_CFLAGS) -fbranch-probabilities -DTESTING -c pre_gen/mp_all.c -o mp_all.o
        $(AR) $(ARFLAGS) $(LIBNAME) mp_all.o
```

Reply:
Yes, or call it tommath_generated.c instead of mp_all.c. this is not a good name due to the helper.pl wildcards.

Reply:
I'll add it to the PR

Reply:
Why do we need the pre_gen directory? Maybe this can be simplified. I take a look later.

Reply:
I already thought if it wouldn't make a bit more sense to have a folder structure with all the source files as in tfm

Reply:
Yes for larger projects I do that usually. But in this case I think it is nice to have the source in a flat hierarchy to browse it directly in github. The utilities and tests should be separate however. In particular if we have a relatively clean repository and rename demo->test, etc->util, remove logs/ etc. It is also a kind of accessibility statement - that the library has a simple structure. It is not a deeply nested project with subsystems and bells and whistles.